sklearn/preprocessing/_encoders.py

```diff
@@ -0,0 +1,888 @@
+    sklearn.preprocessing.LabelBinarizer : binarizes labels in a one-vs-all
+      fashion.
+    sklearn.preprocessing.MultiLabelBinarizer : transforms between iterable of
+      iterables and a multilabel format, e.g. a (samples x classes) binary
+      matrix indicating the presence of a class label.
+    """
+
+    def __init__(self, n_values=None, categorical_features=None,
+                 categories=None, sparse=True, dtype=np.float64,
+                 handle_unknown='error'):
+        self._categories = categories
+        if categories is None:
+            self.categories = 'auto'
+        else:
+            self.categories = categories
+        self.sparse = sparse
+        self.dtype = dtype
+        self.handle_unknown = handle_unknown
+
+        if n_values is not None:
+            pass
+            # warnings.warn("Deprecated", DeprecationWarning)
+        else:
+            n_values = "auto"
+        self._deprecated_n_values = n_values
+
+        if categorical_features is not None:
+            pass
+            # warnings.warn("Deprecated", DeprecationWarning)
+        else:
+            categorical_features = "all"
+        self._deprecated_categorical_features = categorical_features
+
+    # Deprecated keywords
+
+    @property
+    def n_values(self):
+        warnings.warn("The 'n_values' parameter is deprecated.",
+                      DeprecationWarning)
+        return self._deprecated_n_values
+
+    @n_values.setter
+    def n_values(self, value):
+        warnings.warn("The 'n_values' parameter is deprecated.",
+                      DeprecationWarning)
+        self._deprecated_n_values = value
+
+    @property
+    def categorical_features(self):
+        warnings.warn("The 'categorical_features' parameter is deprecated.",
+                      DeprecationWarning)
+        return self._deprecated_categorical_features
+
+    @categorical_features.setter
+    def categorical_features(self, value):
+        warnings.warn("The 'categorical_features' parameter is deprecated.",
+                      DeprecationWarning)
+        self._deprecated_categorical_features = value
+
+    # Deprecated attributes
+
+    @property
+    def active_features_(self):
+        check_is_fitted(self, 'categories_')
+        warnings.warn("The 'active_features_' attribute is deprecated.",
+                      DeprecationWarning)
+        return self._active_features_
+
+    @property
+    def feature_indices_(self):
+        check_is_fitted(self, 'categories_')
+        warnings.warn("The 'feature_indices_' attribute is deprecated.",
+                      DeprecationWarning)
+        return self._feature_indices_
+
+    @property
+    def n_values_(self):
+        check_is_fitted(self, 'categories_')
+        warnings.warn("The 'n_values_' attribute is deprecated.",
+                      DeprecationWarning)
+        return self._n_values_
+
+    def _handle_deprecations(self, X):
+
+        user_set_categories = False
```

Comment:
I would remove this variable, it's used once and declaring it seems to make the code longer.